File: server/src/main/java/io/github/eddiediamondfire/economyplus/listener/EconomyListener.java
package io.github.eddiediamondfire.economyplus.listener;

import io.github.eddiediamondfire.economyplus.Main;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public class EconomyListener implements Listener {
    private final Main plugin;
    public EconomyListener(Main plugin){
        this.plugin = plugin;
    }

    @EventHandler
    public void onJoinEvent(PlayerJoinEvent event){
        Player player = event.getPlayer();

        if(plugin.getAccountManager().accountExist(player.getUniqueId())){
            plugin.getAccountManager().addAccount(player.getUniqueId());
        }
    }

    @EventHandler
    public void onLeaveEvent(PlayerQuitEvent event){
        Player player = event.getPlayer();
    }

}

File: server/src/main/kotlin/io/github/eddiediamondfire/economyplus/listener/EconomyListener.kt
package io.github.eddiediamondfire.economyplus.listener

import io.github.eddiediamondfire.economyplus.Main
import io.github.eddiediamondfire.economyplus.account.AccountManager
import io.github.eddiediamondfire.economyplus.utils.MessageManager
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent
import java.util.*
class EconomyListener(pluginClass: Main) : Listener {

    private var plugin: Main = pluginClass
    private var accountManager: AccountManager = plugin.accountManager
    private var messageManager: MessageManager = plugin.messageManager

    @EventHandler
    fun onPlayerJoinEvent(event: PlayerJoinEvent) {
        val player:Player = event.player
        val playerUUID: UUID = player.uniqueId

        if(accountManager.accountExist(playerUUID))
        {
            val playerName: String = player.displayName
            val uuid: String = player.uniqueId.toString()
            messageManager.sendDebugMessage(ChatColor.YELLOW, "Adding player $playerName with an ID of $uuid to the Player Base")
            accountManager.addAccount(playerUUID)
        }
    }

    @EventHandler
    fun onPlayerLeaveEvent(event: PlayerQuitEvent) {
        val player: Player = event.player
    }
}